homeassistant/config_entries.py

```diff
@@ -438,7 +466,13 @@ def async_entries(self, domain: Optional[str] = None) -> List[ConfigEntry]:
                 # New in 0.79
                 connection_class=entry.get('connection_class',
                                            CONN_CLASS_UNKNOWN))
-            for entry in config['entries']]
+
+            migrated = await entry.async_migrate(self.hass)
```

Comment:
We shouldn't migrate during load, we should migrate when we try to set up the config entry. In the future we might want to add the option to disable config entries, those should not be migrated. When done during setup, we can fail setup if migration failed, we can't fail the load and break the whole system.

Reply:
Sounds good I’ll move it to setup

Reply:
Moved and added new `ENTRY_STATE_MIGRATION_ERROR` for when it fails